test/distributed/_composable/test_compose.py

```diff
@@ -171,6 +179,37 @@ def test_checkpoint_fsdp_submodules_with_param_no_shard(self):
             self._test_parity,
         )
 
+    def test_composable_fsdp_replicate(self):
+        model = CompositeModel(device=torch.device("cpu"))
+        fully_shard(model.l1)
+        with self.assertRaisesRegex(
+            AssertionError, "Cannot apply .*replicate"
+        ):
+            replicate(model.l1)
+        replicate(model.l2)  # should not raise
+
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_in_replicate(self):
+        model = CompositeModel(device=torch.device("cuda"))
+        base_model = copy.deepcopy(model)
+        test_model = copy.deepcopy(model)
+        fully_shard(test_model.l1)
```

Comment:
Should we tested nested/wrapped FSDP as well?

Reply:
@rohan-varma do you mean test `replicate` with FSDP wrapper?

Reply:
SOrry should've clarified, meant passing a policy like @awgu suggests

Reply:
+1 on adding the test, but if there are gaps there, we don't have to fix it in this PR. Having a test with `assertRaisesRegex` will be sufficient for this one, and we can fix in follow up PRs.